### PR TITLE
Add create and destroy scripts for the module to run

### PR DESCRIPTION
**Pull request by taylorludwig**
Will use for implementation in https://github.com/terraform-google-modules/terraform-google-project-factory/pull/343

### CODE REVIEW

**Comment by taylorludwig** [comment]
We might need to add variables for `script_triggers` and `script_depends_on` to replicate this functionality 

https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/modules/core_project_factory/main.tf#L201-L208

**Comment by morgante** [comment]
> We might need to add variables for script_triggers and script_depends_on to replicate this functionality

Yes, let's add `script_triggers` and `module_depends_on` variables.

**Comment by taylorludwig** [comment]
I added `create_script_triggers` and `create_script_arguments` 

You can see the implementation of it used in the project factory PR here 

https://github.com/terraform-google-modules/terraform-google-project-factory/pull/343/files#diff-d3740c193b3aa6f052c215e40f1a18ddR228-R232

I couldn't add a `module_depends_on` because the `local_exec` in this module already has some needed `depends_on` resources (eg gcloud needs to be installed before script execution). And we couldn't merge this module's `depends_on` with the `module_depends_on` variable because terraform would complain that it needed to be a static list. 

However, I was able to replicate the dependency tree by passing extra keys to the `triggers` variable. This forces the same dependency implicitly so we get the same result. You can see in the project factory build the script execution happens at the right time.

File: examples/simple_example/main.tf [hunk 1]
```diff
@@ -26,4 +26,7 @@ module "cli" {
 
   create_cmd_body  = "services enable youtube.googleapis.com --project ${var.project_id}"
   destroy_cmd_body = "services disable youtube.googleapis.com --project ${var.project_id}"
+
+  create_script  = "${path.module}/scripts/script.sh"
```
Comment (morgante) [inline examples/simple_example/main.tf]: We should make a script a separate example from a simple command (though I'm actually curious as to whether scripts could simply be passed as a command?)

Comment (taylorludwig) [reply examples/simple_example/main.tf]: I've added a `script_example` and test. Similar to the `simple_example` but using a script to enable and disable an api. 

I could probably remove the new `script` vars in favor of utilizing the existing `create|destroy_cmd_entrypoint` variables. 

Right now they default to `gcloud` - but passing in the script instead should work (assuming I also prepend the bin directory to the path). Arguments to the script could use the existing `create|destroy_cmd_body` vars. 

Something like this uses all existing vars and will achieve the same effect without breaking original functionality. 

```hcl
module "cli" {
  source = "../.."

  platform              = "linux"
  additional_components = ["kubectl", "beta"]

  create_cmd_entrypoint = "${path.module}/scripts/script.sh"
  create_cmd_body       = "enable ${var.project_id}"

  destroy_cmd_entrypoint = "${path.module}/scripts/script.sh"
  destroy_cmd_body       = "disable ${var.project_id}"
}

```

Comment (taylorludwig) [reply examples/simple_example/main.tf]: I went ahead and committed that. It does make the change quite a bit simpler. I'll update the `project-factory` PR for this approach too

**Comment by morgante** [comment]
@taylorludwig Looks great thanks, released as v0.3.0 so please update pfactory to that.

**Pull request merged**